harness/determined/pytorch/_reducer.py

```diff
@@ -508,5 +494,25 @@ def reduce_metrics(self, for_training: bool) -> Dict[str, Any]:
                         "a dict) if you wish to return a single named metric."
                     )
                 metrics[wrapped.name] = reduced
+        return metrics
+
+    def reduce_metrics(self, for_training: bool) -> Dict[str, Any]:
```

Comment:
Why do we need to break up the reduce_metrics() call for this feature?

I'm not saying it's a bad idea, but I am having trouble seeing why.

Reply:
My rationale was that the reduce_metrics function expects to use all_gather function, but we really just need gather in our case as chief worker will be aggregating the gathered metrics and report to master.

Reply:
I see...

The reason why we used allgather was that typically user code is interested in knowing what the metrics were in some kind of callback, and we wanted to support those kinds of callbacks on more than just the chief worker.

FWIW, I've never encountered a user who was concerned about the additional communications introduced by the allgather.

I don't need you to change it here, but it is something to think about I think.

Reply:
That is a good point, thank you for pointing that out. Right now, we do not support call back that requires metrics, but when we do, we can update accordingly.